### PR TITLE
Update CI test matrix, tests cleanup, add ColorDist tests

### LINKED ISSUE #317: Modify unit tests to use Astropy pytest framework
This is a follow-up for after #313 is merged.

### LINKED ISSUE #554: astropy dev and Python 2 support
Astropy dev has dropped Python 2 support as of astropy/astropy#6556 . CI setup here will need to be updated to avoid chronic failure. What is Ginga's plan going forward?

1. Drop Python 2 support as well.
2. Python 2 version will be in a separate branch with `astropy<3` pinned in `setup.py`.

### PR DESCRIPTION
**Pull request by pllim**
Fix #317 

Fix #554 

Close #346 

TODO:
- [x] Check log every build; e.g., is the `numpy=prerelease` build log still funky?
- [x] Follow up on astropy/ci-helpers#238
- [x] Follow up on astropy/ci-helpers#239

### CODE REVIEW

**Comment by pllim** [comment]
@ejeschke , the `numpy=prerelease` weird log aside (which I don't think is our problem here given Astropy has the same weird log in that setting), this is ready for review. I am hoping this can be merge as soon as possible, so other PRs can benefit from the improved tests.

**Comment by pllim** [comment]
I have sorted out the remaining issues about the weird test log. Everything is fine now. @ejeschke , is this okay to merge? Please advise.

File: setup.cfg [hunk 1]
```diff
@@ -7,8 +7,8 @@ all_files = 1
 upload-dir = doc/_build/html
 show-response = 1
 
-[pytest]
-minversion = 2.2
+[tool:pytest]
+minversion = 3
```
Comment (ejeschke) [inline setup.cfg]: @pllim, astropy-helpers no longer includes its own version of `pytest` right?  I am ok with switching from the default `unittest` to `pytest`, but is this declaration here sufficient to make it a requirement, or does it become a "recommended" 
 install?

File: .travis.yml [hunk 2]
```diff
@@ -50,52 +55,42 @@ matrix:
...
-        # Try Astropy LTS version (disabled for now)
-        #- python: 2.7
-        #  env: ASTROPY_VERSION=lts
-        #- python: 3.5
-        #  env: ASTROPY_VERSION=lts
-
-        # Try older numpy versions
+        # Try older Numpy versions
+        # TODO: Pin Python 2.7 to ASTROPY_VERSION=lts after Astropy v3 release
         - python: 2.7
```
Comment (ejeschke) [inline .travis.yml]: Is it fairly simple to set different supporting package requirements for different python versions?  In particular, is is possible to say that, e.g. for python 2.7 it is ok for numpy>=1.7 while for python 3.x we want numpy>=1.12?

File: ginga/tests/test_colordist.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+"""Test ColorDist.py"""
+from __future__ import division
+
+import numpy as np
+import pytest
+
+from ginga import ColorDist as cd
+
+
+# Some of the code is based on https://github.com/ejeschke/ginga/pull/346
```
Comment (ejeschke) [inline ginga/tests/test_colordist.py]: Do pull requests live on forever, even after they've merged or deleted?  Not sure this comment URL will be valid in the future.

File: setup.cfg (hunk 1 above)
Comment (pllim) [reply setup.cfg]: Yes, astropy removed pytest from its bundled packages. The specified version is the *minimal required* version; see https://docs.pytest.org/en/latest/customize.html#confval-minversion

File: .travis.yml (hunk 2 above)
Comment (pllim) [reply .travis.yml]: This particular file only controls Travis CI testing. It does not stop people from installing weird version combo. Theoretically, the python + numpy combo would be resolved upstream already when people install astropy, which is a dependency of Ginga, so we don't have to worry about it? @bsipocz , do you have anything to add?

File: ginga/tests/test_colordist.py (hunk 3 above)
Comment (pllim) [reply ginga/tests/test_colordist.py]: Yes, merged PR still exists in your repo until GitHub implodes (touch wood). In fact, deleted branches are not really deleted, they are just very hard to access unless you remember the commit hash and/or branch name.

File: .travis.yml (hunk 2 above)
Comment (bsipocz) [reply .travis.yml]: Maybe you can also somehow declare these version requirement in ``setup.py``, but I'm not sure without asking google/SO how to do that conditionally.

Also, for astropy 2.0.x, you can't have numpy 1.7, it's np >=1.9, while 3.0 will be >=1.10.

**Pull request merged**